### PR TITLE
Auth Logout Handler

### PR DESCRIPTION
**Pull request by wild-endeavor**
This adds a logout handler for the auth component.  It will clear the two encrypted cookies used in the oauth flow by setting the expiration to something in the past.

The endpoint also takes the same `redirect_url` query parameter.
To use:
`http://localhost:8088/logout?redirect_url=http://www.google.com`

If there is no redirect url, it just returns a 200 with an empty body.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,7 +25,7 @@ linux_compile:
 
 .PHONY: server
 server:
-	go run cmd/main.go --logtostderr --server.kube-config ~/.kube/config  --config flyteadmin_config.yaml serve
+	go run cmd/main.go --logtostderr --server.kube-config ~/.kube/config  --config ~/flyteadmin_config.yaml serve
```
Comment (katrogan) [inline Makefile]: is this change intentional?

File: pkg/auth/cookie_manager.go [hunk 2]
```diff
@@ -93,3 +94,28 @@ func (c CookieManager) SetTokenCookies(ctx context.Context, writer http.Response
 	}
 	return nil
 }
+
+func getLogoutAccessCookie() *http.Cookie {
+	return &http.Cookie{
+		Name:     accessTokenCookieName,
+		Value:    "",
```
Comment (katrogan) [inline pkg/auth/cookie_manager.go]: do you need to initialize these zero values?

File: Makefile (hunk 1 above)
Comment (wild-endeavor) [reply Makefile]: no no, just debugging.

File: pkg/auth/cookie_manager.go (hunk 2 above)
Comment (wild-endeavor) [reply pkg/auth/cookie_manager.go]: no, but may as well make this one explicit.  just because the value is set when you log in, I feel like I like explicitly unsetting it, but that's the irrational part of me talking.

Comment (katrogan) [reply pkg/auth/cookie_manager.go]: okie dokes

File: pkg/auth/handlers.go [hunk 3]
```diff
@@ -252,15 +251,18 @@ func GetMetadataEndpointRedirectHandler(ctx context.Context, authCtx interfaces.
 	}
 }
 
-// These are here for CORS handling. Actual serving of the OPTIONS request will be done by the gorilla/handlers package
-type CorsHandlerDecorator func(http.Handler) http.Handler
+func GetLogoutEndpointHandler(ctx context.Context, authCtx interfaces.AuthenticationContext) http.HandlerFunc {
+	return func(writer http.ResponseWriter, request *http.Request) {
+		logger.Debugf(ctx, "Deleting auth cookies")
```
Comment (schottra) [inline pkg/auth/handlers.go]: Just out of curiosity, are we in the habit of leaving in debug statements? They don't blog our staging/production logs right?

File: pkg/auth/handlers.go [hunk 4]
```diff
@@ -252,15 +251,18 @@ func GetMetadataEndpointRedirectHandler(ctx context.Context, authCtx interfaces.
...
+		// Redirect if one was given
+		queryParams := request.URL.Query()
+		if redirectURL := queryParams.Get(RedirectURLParameter); redirectURL != "" {
+			http.Redirect(writer, request, redirectURL, http.StatusTemporaryRedirect)
+		}
+	}
 }
+
+// These are here for CORS handling. Actual serving of the OPTIONS request will be done by the gorilla/handlers package
+type CorsHandlerDecorator func(http.Handler) http.Handler
```
Comment (schottra) [inline pkg/auth/handlers.go]: Is this still needed?

File: pkg/rpc/adminservice/base.go [hunk 5]
```diff
@@ -57,7 +57,7 @@ func NewAdminServer(kubeConfig, master string) *AdminService {
 	defer func() {
 		if err := recover(); err != nil {
 			adminScope.MustNewCounter("initialization_panic",
-				"panics encountered initializating the admin service").Inc()
```
Comment (schottra) [inline pkg/rpc/adminservice/base.go]: I like this word better. :-p

File: pkg/auth/handlers.go (hunk 4 above)
Comment (wild-endeavor) [reply pkg/auth/handlers.go]: It's not but I'm gonna remove it later.  Don't want to get more approvals.  Already dismissed this approval enough times

File: pkg/auth/handlers.go (hunk 3 above)
Comment (wild-endeavor) [reply pkg/auth/handlers.go]: I am yeah, it helps with debugging.  The logging level is controlled through a runtime config so we can play with it on the fly.
